sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -264,15 +268,31 @@ private[hive] object HadoopTableReader extends HiveInspectors {
    * @param nonPartitionKeyAttrs Attributes that should be filled together with their corresponding
    *                             positions in the output schema
    * @param mutableRow A reusable `MutableRow` that should be filled
+   * @param convertdeserializer The `Deserializer` covert the `deserializer`                
    * @return An `Iterator[Row]` transformed from `iterator`
    */
   def fillObject(
       iterator: Iterator[Writable],
       deserializer: Deserializer,
       nonPartitionKeyAttrs: Seq[(Attribute, Int)],
-      mutableRow: MutableRow): Iterator[Row] = {
+      mutableRow: MutableRow,
+      convertdeserializer: Option[Deserializer] = None): Iterator[Row] = {
```

Comment:
Instead of passing the `deserializer`, how about take the `converter` as the argument? By the way, I think Hive provides the `IdentityConverter`, which mean we can make the parameter as "ObjectInspectorConverters.Converter", not necessary wrapped by `Option`.